grizli/jwst_utils.py

```diff
@@ -122,11 +122,15 @@ def img_with_wcs(input):
     # from jwst.assign_wcs import assign_wcs
 
     # HDUList -> jwst.datamodels.ImageModel
-    
+
     # Generate WCS as image
+    if not isinstance(input, pyfits.HDUList):
+        input = pyfits.open(input)
     if isinstance(input, pyfits.HDUList):
         if input[0].header['INSTRUME'] == 'NIRISS':
             if input[0].header['FILTER'].startswith('GR'):
+                input[0].header['FILTER_INFO'] = input[0].header['FILTER'] # just to save it so we can change it back at the end
```

Comment:
FITS header keywords generally need to be 8 characters or less, though I'm not too careful with that elsewhere.  Maybe `OFILTER` and `OEXPTYPE`? (those are "Oh's", not zeros)